kernel/src/debug/gdbstub.rs

```diff
@@ -306,25 +303,27 @@ pub mod svsm_gdbstub {
         });
     }
 
-    struct GdbStubConnection;
+    struct GdbStubConnection<'a> {
+        serial_port: SerialPort<'a>,
+    }
 
-    impl GdbStubConnection {
-        const fn new() -> Self {
-            Self {}
+    impl GdbStubConnection<'_> {
+        fn new(platform: &'static dyn SvsmPlatform) -> Self {
```

Comment:
```suggestion
    impl<'a> GdbStubConnection<'a> {
        fn new(platform: &'a dyn SvsmPlatform) -> Self {
```

Or just change the lifetime of the `serial_port` field to `'static`.

Reply:
Since `GdbStubConnection` is already typed with a lifetime, adding the same lifetime to the `& dyn SvsmPlatform` is an easy change.